assets/js/custom/Translation.js

```diff
@@ -2,6 +2,137 @@
 
 // eslint-disable-next-line no-unused-vars
 function Translation (hasDescription, hasCredit) {

```

Comment:
I think you can inject the `creditText` in the constructor and in https://github.com/danielzhang130/Catroweb/blob/01693a76e9b0be7cdabc24669557441018c029f8/templates/Program/program.html.twig#L305
you can do 
```
new Translation(
      {% if program.description %} true, {% else %} false, {% endif %}
      {% if program.credits %} true, {% else %} false, {% endif %},
      '{{ 'programs.reactionsText'|trans({}, 'catroweb') }}'
    )
```

Reply:
Added translatedByText in constructor